### PR TITLE
VR > Add the customizer Text section for views V2

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://moderntribe.atlassian.net/browse/TEC-3068
🎫 https://moderntribe.atlassian.net/browse/TEC-3070
🎫 https://moderntribe.atlassian.net/browse/TEC-3071
🎥 https://www.loom.com/share/32e7446d83ae4b5ebe128e5543df3b64

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -217,6 +217,9 @@ Still not happy? Shoot us an email to support@theeventscalendar.com or tweet to
 
 = [5.0.1] TBD =
 
+* Feature: Add the "Text" section in the customizer, for the new views. [TEC-3068]
```
Comment (bordoni) [inline readme.txt]:
```suggestion
* Feature - Add the "Text" section in the customizer, for the new views. [TEC-3068]
```

File: readme.txt [hunk 2]
```diff
@@ -217,6 +217,9 @@ Still not happy? Shoot us an email to support@theeventscalendar.com or tweet to
...
+* Feature: Add the "Primary Text Color" option in the customizer, for the new views. [TEC-3070]
```
Comment (bordoni) [inline readme.txt]:
```suggestion
* Feature - Add the "Primary Text Color" option in the customizer, for the new views. [TEC-3070]
```

File: readme.txt [hunk 3]
```diff
@@ -217,6 +217,9 @@ Still not happy? Shoot us an email to support@theeventscalendar.com or tweet to
...
+* Feature: Add the "Secondary Text Color" option in the customizer, for the new views. [TEC-3071]
```
Comment (bordoni) [inline readme.txt]:
```suggestion
* Feature - Add the "Secondary Text Color" option in the customizer, for the new views. [TEC-3071]
```

File: src/Tribe/Customizer/Text.php [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+ * @since   TBD
+ */
+
+// Don't load directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+final class Tribe__Events__Customizer__Text extends Tribe__Customizer__Section {
+
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: * :no_entry_sign: Missing doc comment for class Tribe__Events__Customizer__Text
* :no_entry_sign: Class name is not valid; consider Tribe_Events_Customizer_Text instead

<sub>hash: b7db5bddac4b5a087c8f676ed1fdb5c5</sub>

File: src/Tribe/Customizer/Text.php [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+		return tribe( 'tec.customizer.text' );
+	}
+
+	/**
+	 * Grab the CSS rules template.
+	 * 
+	 * @since TBD
+	 *
+	 * @param string  $template
+	 * @return string $template
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: * :no_entry_sign: Missing parameter comment
* :no_entry_sign: Expected 1 spaces after parameter type; 2 found

<sub>hash: d507d18b34229d0b3933a1ae3634b5e9</sub>

File: src/Tribe/Customizer/Text.php [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+	 */
+	public function get_css_template( $template ) {
+		$customizer = Tribe__Customizer::instance();
+		$settings   = $customizer->get_option( [ $this->ID ] );
+		if ( $customizer->has_option( $this->ID, 'primary_text_color' ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: ad4e023a4481307f91f00af731a4e2f4</sub>

File: src/Tribe/Customizer/Text.php [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+			$primary_text_color     = new Tribe__Utils__Color( $settings['primary_text_color'] );
+			$primary_text_color_rgb = $primary_text_color::hexToRgb( $settings['primary_text_color'] );
+			
+			$template .= '
+				.tribe-common .tribe-common-b1,
+				.tribe-common .tribe-common-b2,
+				.tribe-common .tribe-common-b3 {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: a20208ebd69f312cd95a2e82e4586f67</sub>

File: src/Tribe/Customizer/Text.php [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+				.tribe-common .tribe-common-h1,
+				.tribe-common .tribe-common-h2,
+				.tribe-common .tribe-common-h3,
+				.tribe-common .tribe-common-h4,
+				.tribe-common .tribe-common-h5,
+				.tribe-common .tribe-common-h6,
+				.tribe-common .tribe-common-h7,
+				.tribe-common .tribe-common-h8 {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 3d20e4542e6b79a834dfb2914a229f74</sub>

File: src/Tribe/Customizer/Text.php [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+				.tribe-common a:visited,
+				.tribe-common .tribe-common-c-btn-border:hover,
+				.tribe-common .tribe-common-c-btn-border:focus,
+				.tribe-common a.tribe-common-c-btn-border:active, 
+				.tribe-common a.tribe-common-c-btn-border:focus, 
+				.tribe-common a.tribe-common-c-btn-border:hover,
+				.tribe-theme-twentyseventeen .tribe-common a:hover, 
+				.tribe-theme-twentyseventeen .tribe-common a:focus {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 607d521e9c8bacf4e9ab85fc788cc470</sub>

File: src/Tribe/Customizer/Text.php [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+			';
+
+			$template .= '
+				.tribe-common .tribe-common-form-control-text__input,
+				.tribe-common--breakpoint-medium.tribe-common .tribe-common-form-control-text__input {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 0aa8b189248b11caaad02577e9cef670</sub>

File: src/Tribe/Customizer/Text.php [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+			';
+
+			$template .= '
+				.tribe-events .tribe-events-c-view-selector__list-item-text,
+				.tribe-common--breakpoint-medium.tribe-events .tribe-events-c-view-selector--labels .tribe-events-c-view-selector__button-text {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 7d8258898f1813a1fa763fa1c95f3855</sub>

File: src/Tribe/Customizer/Text.php [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+			';
+				
+			$template .= '
+				.tribe-events .datepicker .dow,
+				.tribe-events .datepicker .day,
+				.tribe-events .datepicker .month, 
+				.tribe-events .datepicker .year,
+				.tribe-events .datepicker .datepicker-switch {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 8015582b282f8e77e47527877513bc87</sub>

File: src/Tribe/Customizer/Text.php [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+			';
+			
+			$template .= '
+				.tribe-common .tribe-common-anchor-thin:active, 
+				.tribe-common .tribe-common-anchor-thin:focus, 
+				.tribe-common .tribe-common-anchor-thin:hover {
+					border-bottom-color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 8c0b2c7a95121229acef31cde2a68fde</sub>

File: src/Tribe/Customizer/Text.php [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+			';
+			
+			$template .= '
+				.tribe-events .tribe-events-calendar-month__day-date-link,
+				.tribe-events .tribe-events-calendar-month__calendar-event-tooltip-datetime {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: a0bcaee39e299b5c37b0f196b5ed9a1d</sub>

File: src/Tribe/Customizer/Text.php [hunk 15]
```diff
@@ -0,0 +1,272 @@
...
+			';
+			
+			// PRO Styles.
+			$template .= '
+				.tribe-events-pro .tribe-events-pro-week-grid__events-time-tag,
+				.tribe-events-pro .tribe-events-pro-week-grid__event-tooltip-datetime,
+				.tribe-events-pro .tribe-events-pro-week-grid__header-column--current .tribe-events-pro-week-grid__header-column-weekday {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: b6c243f1171c59f60c29562b5f917df3</sub>

File: src/Tribe/Customizer/Text.php [hunk 16]
```diff
@@ -0,0 +1,272 @@
...
+			';
+		
+			$template .= '
+				.tribe-events-pro.tribe-events-view--week .datepicker .day.disabled {
+					color: <%= text.primary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 56c0b933f24cf7c0a55729ad8b61b478</sub>

File: src/Tribe/Customizer/Text.php [hunk 17]
```diff
@@ -0,0 +1,272 @@
...
+			';
+
+		}
+
+		if ( $customizer->has_option( $this->ID, 'secondary_text_color' ) ) {
+			$template .= '
+				.tribe-common .tribe-common-form-control-toggle__label,
+				.tribe-events .tribe-events-calendar-list__event-date-tag-weekday {
+					color: <%= text.secondary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 19938dc35d0bf62e1a8b861300b291ff</sub>

File: src/Tribe/Customizer/Text.php [hunk 18]
```diff
@@ -0,0 +1,272 @@
...
+			';
+
+			$template .= '
+				.tribe-common .tribe-common-c-btn-border, 
+				.tribe-common a.tribe-common-c-btn-border {
+					color: <%= text.secondary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 4d92e411f0aceab37cec7fa24804fd7a</sub>

File: src/Tribe/Customizer/Text.php [hunk 19]
```diff
@@ -0,0 +1,272 @@
...
+			';
+
+			$template .= '
+				.tribe-common .tribe-common-form-control-text__input::placeholder {
+					color: <%= text.secondary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: eae4fda9d0ac8d1f95c0213782804a32</sub>

File: src/Tribe/Customizer/Text.php [hunk 20]
```diff
@@ -0,0 +1,272 @@
...
+			';
+
+			$template .= '
+				.tribe-common--breakpoint-medium.tribe-events .tribe-events-c-nav__prev, 
+				.tribe-common--breakpoint-medium.tribe-events .tribe-events-c-nav__next {
+					color: <%= text.secondary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: e88f4e7e1f62f808566ede20648592f0</sub>

File: src/Tribe/Customizer/Text.php [hunk 21]
```diff
@@ -0,0 +1,272 @@
...
+
+			// PRO Styles.
+			$template .= '
+				.tribe-events .tribe-events-calendar-month__calendar-event-datetime,
+				.tribe-events-pro .tribe-events-pro-photo__event-date-tag-month,
+				.tribe-events-pro .tribe-events-pro-week-grid__event-datetime,
+				.tribe-events-pro .tribe-events-pro-map__event-date-tag-month,
+				.tribe-common--breakpoint-medium.tribe-events-pro .tribe-events-pro-map__event-distance {
+					color: <%= text.secondary_text_color %>;
+				}
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 36d8b17ed163407571da7e240885a039</sub>

File: src/Tribe/Customizer/Text.php [hunk 22]
```diff
@@ -0,0 +1,272 @@
...
+	/**
+	 * Setup the Customizer section.
+	 * 
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function setup() {
+		$this->defaults = [
+			'primary_text_color'   => '#141827',
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 79dbc8d093cd32cd8698e4d2fab4454f</sub>

File: src/Tribe/Customizer/Text.php [hunk 23]
```diff
@@ -0,0 +1,272 @@
...
+			'secondary_text_color' => '#727272',
+		];
+
+		$this->arguments = [
+			'priority'    => 60,
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 75224f420cbd264f6f5b7fd5aaaa2df6</sub>

File: src/Tribe/Customizer/Text.php [hunk 24]
```diff
@@ -0,0 +1,272 @@
...
+			'description' => esc_html__( 'Options selected here will override what was selected in the General section.', 'the-events-calendar' ),
+		];
+	}
+
+	/**
+	 * Create the Fields/Settings for this sections.
+	 *
+	 * @param  WP_Customize_Section $section The WordPress section instance.
+	 * @param  WP_Customize_Manager $manager [description]
+	 *
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Parameter comment must end with a full stop

<sub>hash: eb863842f566e9841ef21977f52b6e1a</sub>

File: src/Tribe/Customizer/Text.php [hunk 25]
```diff
@@ -0,0 +1,272 @@
...
+	 * @return void
+	 */
+	public function register_settings( WP_Customize_Section $section, WP_Customize_Manager $manager ) {
+		$customizer = Tribe__Customizer::instance();
+
+		$manager->add_setting(
+			$customizer->get_setting_name( 'primary_text_color', $section ),
+			[
+				'default'              => $this->get_default( 'primary_text_color' ),
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 75be51964582ab078a6778ed8ff0c8d0</sub>

File: src/Tribe/Customizer/Text.php [hunk 26]
```diff
@@ -0,0 +1,272 @@
...
+				'sanitize_js_callback' => 'maybe_hash_hex_color',
+			]
+		);
+
+		$manager->add_control(
+			new WP_Customize_Color_Control(
+				$manager,
+				$customizer->get_setting_name( 'primary_text_color', $section ),
+				[
+					'label'   => esc_html__( 'Primary Text Color', 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 1c200cfdbcc0e75885e7cab9b52ac41f</sub>

File: src/Tribe/Customizer/Text.php [hunk 27]
```diff
@@ -0,0 +1,272 @@
...
+					'section' => $section->id,
+				]
+			)
+		);
+
+		$manager->add_setting(
+			$customizer->get_setting_name( 'secondary_text_color', $section ),
+			[
+				'default'              => $this->get_default( 'secondary_text_color' ),
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: d9452cedcc74d48dabda4cefd4df3b76</sub>

File: src/Tribe/Customizer/Text.php [hunk 28]
```diff
@@ -0,0 +1,272 @@
...
+				'sanitize_js_callback' => 'maybe_hash_hex_color',
+			]
+		);
+
+		$manager->add_control(
+			new WP_Customize_Color_Control(
+				$manager,
+				$customizer->get_setting_name( 'secondary_text_color', $section ),
+				[
+					'label'   => esc_html__( 'Secondary Text Color', 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Text.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 9b71f26416fcf3d46e14a676aaee0944</sub>

**Pull request merged**